packages/backend/test/peripherals/database/SyncStatusRepository.test.ts

```diff
@@ -0,0 +1,51 @@
+import { expect } from 'chai'
+
+import {
+  Store,
+  SyncStatusRepository,
+} from '../../../src/peripherals/database/SyncStatusRepository'
+import { mock } from '../../mock'
+
+describe(SyncStatusRepository.name, () => {
+  it('writes value to store', async () => {
+    let key = ''
+    let value = ''
+    const repository = new SyncStatusRepository(
+      mock<Store>({
+        set: async (...args) => void ([key, value] = args),
+      })
+    )
+
+    await repository.setLastBlockNumberSynced(10)
+
+    expect(key).to.eq('lastBlockNumberSynced')
+    expect(value).to.eq('10')
+  })
+
+  it('reads value from store', async () => {
+    let key = ''
+    const repository = new SyncStatusRepository(
+      mock<Store>({
+        async get(k) {
+          key = k
+          return '20'
+        },
+      })
+    )
+
+    const actual = await repository.getLastBlockNumberSynced()
+
+    expect(actual).to.eq(20)
+    expect(key).to.eq('lastBlockNumberSynced')
```

Comment:
You can actually move this expect to the first line of the `async get(k) {` function so that there is no need for `let key = ''`

Reply:
What if `get` is never called?

Reply:
Then how the value 20 is being returned?

Reply:
It could be hardcoded. We're just running one example.

Reply:
Ok let's leave it like this then